File: M4/construction/construction/report/boq_billing_progress_report/boq_billing_progress_report.js
// Copyright (c) 2016, Nxweb and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["BOQ Billing Progress Report"] = {
		"filters": [
	     {
			"fieldname":"project",
			"label": __("Project"),
			"fieldtype": "Link",
			"width": "80",
			"options":"Project"
		  },
		  {
			"fieldname":"project_structure",
			"label": __("Project structure"),
			"fieldtype": "Link",
			"width": "80",
			"options":"Project Structure"
		  },
		  {
			"fieldname":"billing_status",
			"label": __("Billing Status"),
			"fieldtype": "Select",
			"width": "80",
			"options":"\nTo Quotation\nTo Order\nOrdered\nNot Billable"
		  }


	],
	"formatter": function(value, row, column, data, default_formatter){
	  	value = default_formatter(value, row, column, data);
	  	if (column.fieldname == "project" && data.billing_status == "To Quotation"){
	  		value = "<span style ='color:RebeccaPurple'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "project_structure" && data.billing_status == "To Quotation"){
	  		value = "<span style ='color:RebeccaPurple'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "item_of_work" && data.billing_status == "To Quotation"){
	  		value = "<span style ='color:RebeccaPurple'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "billing_status" && data.billing_status == "To Quotation"){
	  		value = "<span style ='color:RebeccaPurple'>" + value + "</span>";
	  	}

	  	if (column.fieldname == "project" && data.billing_status == "To Order"){
	  		value = "<span style ='color:Maroon'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "project_structure" && data.billing_status == "To Order"){
	  		value = "<span style ='color:Maroon'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "item_of_work" && data.billing_status == "To Order"){
	  		value = "<span style ='color:Maroon'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "billing_status" && data.billing_status == "To Order"){
	  		value = "<span style ='color:Maroon'>" + value + "</span>";
	  	}


	  	if (column.fieldname == "project" && data.billing_status == "Not Billable"){
	  		value = "<span style ='color:Purple'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "project_structure" && data.billing_status == "Not Billable"){
	  		value = "<span style ='color:Purple'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "item_of_work" && data.billing_status == "Not Billable"){
	  		value = "<span style ='color:Purple'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "billing_status" && data.billing_status == "Not Billable"){
	  		value = "<span style ='color:Purple'>" + value + "</span>";
	  	}


	  	if (column.fieldname == "project" && data.billing_status == "Ordered"){
	  		value = "<span style ='color:green'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "project_structure" && data.billing_status == "Ordered"){
	  		value = "<span style ='color:green'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "item_of_work" && data.billing_status == "Ordered"){
	  		value = "<span style ='color:green'>" + value + "</span>";
	  	}
	  	if (column.fieldname == "billing_status" && data.billing_status == "Ordered"){
	  		value = "<span style ='color:green'>" + value + "</span>";
	  	}

	  	return value;
	  }
};

File: M4/construction/construction/report/boq_billing_progress_report/boq_billing_progress_report.py
# Copyright (c) 2013, Nxweb and contributors
# For license information, please see license.txt

import frappe
from frappe import _

def execute(filters=None):
	filters = frappe._dict(filters or {})
	columns = get_columns(filters)
	data = get_data(filters)
	return columns, data



def get_columns(filters):
	return[
	   {
	   'label': _('Project'),
	   'fieldtype': 'Data',
	   'fieldname': 'project',
	   'width': 230,
	   'options':'Project',
	   },
	   {
	   'label': _('Project Structure'),
	   'fieldtype': 'Data',
	   'fieldname': 'project_structure',
	   'width': 230,
	   'options':'Project Structure',
	   },
	   {
	   'label': _('Item Of Work'),
	   'fieldtype': 'Data',
	   'fieldname': 'item_of_work',
	   'width': 230,
	   'options':'Item of Work',
	   },
	   {
	   'label': _('Billing Status'),
	   'fieldtype': 'Data',
	   'fieldname': 'billing_status',
	   'width': 200,
	   'options':'BOQ',
	   },
	   {
	   'label': _('Billing Percentage'),
	   'fieldtype': ' Percent',
	   'fieldname': 'billing_progress',
	   'width': 200,
	   'options':'BOQ',
	   }

	   ]

def get_data(filters):
	conditions = get_conditions(filters)
	data =[]
	return frappe.db.sql(""" SELECT boq.name,boq.project,boq.project_structure,boq.item_of_work,boq.work_status,boq.billing_status,boq.working_progress,boq.billing_progress

		                     FROM `tabBOQ` boq 

		                     WHERE  boq.docstatus = 1 %s """ %  conditions,filters, as_dict=1)


def get_conditions(filters):
	conditions = ""
	if filters.get("project"):
		conditions +=" and boq.project = %(project)s"
	if filters.get("project_structure"):
		conditions +=" and boq.project_structure = %(project_structure)s"
	if filters.get("item_of_work"):
		conditions +=" and boq.item_of_work = %(item_of_work)s"
	if filters.get("billing_status"):
		conditions +=" and boq.billing_status = %(billing_status)s"
	
	
	
	
	return conditions
